apollo-gradle-plugin/src/main/groovy/com/apollostack/android/gradle/ApolloIRgenTask.groovy

```diff
@@ -0,0 +1,60 @@
+package com.apollostack.android.gradle
+
+import com.google.common.collect.Sets
+import com.moowork.gradle.node.task.NodeTask
+import org.gradle.api.GradleException
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.util.PatternSet
+
+public class ApolloIRgenTask extends NodeTask {
+  private static final String APOLLO_CODEGEN = "node_modules/apollo-codegen/lib/cli.js"
+  static final String NAME = "generate%sApolloIR"
+  List<ApolloExtension> config
```

Comment:
The ApolloIRgenTask is registered per variant and a variant can have multiple sourceSet directories.

For example, the debug variant on android has src/main and src/debug in its sourceSets. So the List of extensions was needed here.